system-addon/lib/TippyTopFeed.jsm

```diff
@@ -0,0 +1,134 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const {utils: Cu} = Components;
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+Cu.importGlobalProperties(["fetch", "URL"]);
+
+const {actionCreators: ac, actionTypes: at} = Cu.import("resource://activity-stream/common/Actions.jsm", {});
+const {PersistentCache} = Cu.import("resource://activity-stream/lib/PersistentCache.jsm", {});
+const {Prefs} = Cu.import("resource://activity-stream/lib/ActivityStreamPrefs.jsm", {});
+
+const DEFAULT_TIPPYTOP_JSON_PATH = "resource://activity-stream/data/content/tippytop/top_sites.json";
+const DEFAULT_TIPPYTOP_URL_PREFIX = "resource://activity-stream/data/content/tippytop/images/";
+const TIPPYTOP_UPDATE_TIME = 24 * 60 * 60 * 1000; // 24 hours
+
+this.getDomain = function(url) {
+  let domain = new URL(url).hostname;
+  if (domain && domain.startsWith("www.")) {
```

Comment:
Do we want to handle those [www2](https://en.wikipedia.org/wiki/WWW2) mirror sites? There are quite a few of them (www1/2/3/4) in my Places.

Reply:
hmm, i guess we could do a regex check for most cases?